core/src/multi_proof_verification.rs

```diff
@@ -0,0 +1,636 @@
+use crate::{
+    multi_proof::{MultiProof, SubPathProof},
+    proof::{
+        hash_path, shared_bits, PathProofTerminal, PathProofVerificationError, VerifiedPathProof,
+    },
+    trie::{InternalData, KeyPath, LeafData, Node, NodeHasher, NodeHasherExt, TERMINATOR},
+};
+use alloc::collections::BTreeMap;
+use bitvec::{order::Msb0, vec::BitVec, view::BitView};
+
+/// A verifiable proof of multiple path through the trie.
+#[derive(Debug, Clone)]
+pub struct VerifiableMultiProof<H: NodeHasher> {
```

Comment:
Why isn't a regular `MultiProof` verifiable? Isn't this just another step along the way to verification?

This seems like it is just exposing an internal step.

Two other thoughts:
  1. It doesn't seem like this format is necessary for verifying a multi-proof.
  2. If (1) is wrong and we do need this format, then why doesn't `multi_proof` just create this format in the first place? It would be simpler than running the super-compact algorithm and then transforming into this.